src/evaluateXPath.ts

```diff
@@ -159,17 +195,20 @@ export type Options = {
  *
  * @returns The result of executing this XPath
  */
-function evaluateXPath(
+const evaluateXPath = function _evaluateXPath<
+	T extends Node,
+	TReturnType extends keyof IReturnTypes
```

Comment:
This is because TypeScript is detecting the type of `evaluateXPath.NODES_TYPE` as `ReturnType`. Declaring the type to be `ReturnType.NODES`, either by casting or declaring it, it does work.

@Martinn1996 if you apply this pattern, it should work:
```
evaluateXPath[
	'BOOLEAN_TYPE'
] = evaluateXPath.BOOLEAN_TYPE = ReturnType.BOOLEAN as ReturnType.BOOLEAN;
```